Can not delete an invalid pipeline and the controller crashes

**Describe the bug**

When a pipeline spec is incorrectly configured with a non-existing `to` vertex in the `edge`, after applying to k8s, the pipeline status shows the validation error, which is expected. However, when attempting to delete the pipeline, the controller crashes with ```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xe43d00]

goroutine 455 [running]:
github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1.Pipeline.ListAllEdges({{{0x1bd366b, 0x8}, {0xc00050e080, 0x1d}}, {{0xc000c2fb50, 0xf}, {0x0, 0x0}, {0xc000c2fb60, 0xf}, ...}, ...})
	/Users/jwang21/workspace/numaproj/numaflow/pkg/apis/numaflow/v1alpha1/pipeline_types.go:92 +0x180
github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1.Pipeline.GetAllBuffers({{{0x1bd366b, 0x8}, {0xc00050e080, 0x1d}}, {{0xc000c2fb50, 0xf}, {0x0, 0x0}, {0xc000c2fb60, 0xf}, ...}, ...})
	/Users/jwang21/workspace/numaproj/numaflow/pkg/apis/numaflow/v1alpha1/pipeline_types.go:140 +0x58
github.com/numaproj/numaflow/pkg/reconciler/pipeline.(*pipelineReconciler).cleanUpBuffers(0xc000325880, {0x23fafd8, 0xc000365020}, 0xc000afe200, 0xedb054510?)
	/Users/jwang21/workspace/numaproj/numaflow/pkg/reconciler/pipeline/controller.go:377 +0x98
github.com/numaproj/numaflow/pkg/reconciler/pipeline.(*pipelineReconciler).reconcile(0x23fafd8?, {0x23fafd8, 0xc000365020}, 0xc000afe200)
	/Users/jwang21/workspace/numaproj/numaflow/pkg/reconciler/pipeline/controller.go:116 +0x846
github.com/numaproj/numaflow/pkg/reconciler/pipeline.(*pipelineReconciler).Reconcile(0xc000325880, {0x23fafd8, 0xc000364ea0}, {{{0xc000c2fb60?, 0x1f978a0?}, {0xc000c2fb50?, 0x30?}}})
	/Users/jwang21/workspace/numaproj/numaflow/pkg/reconciler/pipeline/controller.go:80 +0x536
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0001c2210, {0x23fafd8, 0xc000364e70}, {{{0xc000c2fb60?, 0x1f978a0?}, {0xc000c2fb50?, 0x4045d4?}}})
	/Users/jwang21/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:114 +0x28b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001c2210, {0x23faf30, 0xc000325680}, {0x1e63300?, 0xc000278100?})
	/Users/jwang21/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:311 +0x352
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001c2210, {0x23faf30, 0xc000325680})
	/Users/jwang21/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:266 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/Users/jwang21/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/Users/jwang21/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:223 +0x31c
```




---

<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We often sort issues this way to know what to prioritize.

--- a/pkg/apis/numaflow/v1alpha1/pipeline_types.go
+++ b/pkg/apis/numaflow/v1alpha1/pipeline_types.go
@@ -89,6 +89,9 @@ func (p Pipeline) ListAllEdges() []Edge {
 	for _, e := range p.Spec.Edges {
 		edgeCopy := e.DeepCopy()
 		toVertex := p.GetVertex(e.To)
+		if toVertex == nil {
+			continue
+		}
 		if toVertex.UDF == nil || toVertex.UDF.GroupBy == nil {
 			// Clean up parallelism if downstream vertex is not a reduce UDF.
 			// This has been validated by the controller, harmless to do it here.

